optimum/exporters/onnx/convert.py

```diff
@@ -432,6 +482,53 @@ def export_encoder_decoder_model(
     return outputs
 
 
+def export_decoder_model(
+    model: Union["PreTrainedModel", "TFPreTrainedModel"],
+    config: OnnxConfig,
+    opset: int,
+    decoder_output: Path,
+    decoder_with_past_output: Optional[Path] = None,
+    device: str = "cpu",
+) -> Tuple[List[List[str]], List[List[str]]]:
+    """
+    Exports a Pytorch or TensorFlow encoder decoder model to an ONNX Intermediate Representation.
+    The following method exports the encoder and decoder components of the model as separate
+    ONNX files.
+
+    Args:
+        model ([`PreTrainedModel`] or [`TFPreTrainedModel`]):
+            The model to export.
+        config ([`~exporters.onnx.config.OnnxConfig`]):
+            The ONNX configuration associated with the exported model.
+        opset (`int`):
+            The version of the ONNX operator set to use.
+        decoder_output (`Path`):
+            Directory to store the exported decoder ONNX model.
+        decoder_with_past_output (`Optional[Path]`, defaults to `None`):
+            Directory to store the exported decoder with past ONNX model. Required when `past_key_values` are exported.
+        device (`str`, *optional*, defaults to `cpu`):
+            The device on which the ONNX model will be exported. Either `cpu` or `cuda`. Only PyTorch is supported for
+            export on CUDA devices.
+    Returns:
+        `Tuple[List[List[str]], List[List[str]]]`: A tuple with an ordered list of the model's inputs, and the named
+        inputs from the ONNX configuration.
+    """
+    models_for_export = get_decoder_models_for_export(model, config)
+    outputs = []
+
+    # export decoder
```

Comment:
nit: not needed imo

Reply:
Do you mean the external function is not needed? Or the comment? Or the logic with lists?

Reply:
The comment sorry.

Reply:
right, remove in last commit